distributed/worker.py

```diff
@@ -4363,6 +4366,8 @@ async def apply_function_async(
     try:
         result = await function(*args, **kwargs)
     except Exception as e:
+        # NOTE: we don't catch `BaseException`s because we're running in the main thread,
+        # plus `asyncio.CancelledError` is a `BaseException`
```

Comment:
I would prefer us keeping this all as consistent as possible for all kinds of executors. I am not very fond of the async execution pattern but if it's possible at all, I would like to see the same behaviour